code/game/objects/items/stacks/rods.dm

```diff
@@ -74,3 +74,19 @@
 		DISABLE_BITFIELD(obj_flags, IN_USE)
 		use(4)
 
+/obj/item/stack/rods/attack_turf(turf/T, mob/living/user)
+	if(T.type != /turf/open/floor/plating)
+		to_chat(user, span_warning("You must remove the plating first."))
+		return
+	if(get_amount() < 2)
+		to_chat(user, span_warning("You need more rods."))
+		return
+
+	to_chat(user, span_notice("Reinforcing the floor."))
+	if(!do_after(user, 30, TRUE, src, BUSY_ICON_BUILD) || (T.type != /turf/open/floor/plating))
```

Comment:
Type check should go first so you don't waste people's time waiting for the do_after only for it to silently fail

Reply:
its so if it changes from plating during the do_after